akka-contrib/docs/circuitbreaker.rst

```diff
@@ -0,0 +1,184 @@
+           .copy(
+             failureDetector = {
+               _ match  {
+                 case Response(Left(_)) => true
+                 case _ => false
+               }
+             }
+           )
+           .propsForTarget(potentiallyFailingService),
+         "serviceCircuitBreaker"
+       )
+
+     override def receive: Receive = {
+       case AskFor(requestToForward) =>
+         serviceCircuitBreaker ! Request(requestToForward)
+
+       case Right(Response(content)) =>
+         //handle response
+         log.info("Got successful response {}", content)
+
+       case Response(Right(content)) =>
+         //handle response
+         log.info("Got successful response {}", content)
+
+       case Response(Left(content)) =>
+         //handle response
+         log.info("Got failed response {}", content)
+
+       case CircuitOpenFailure(failedMsg) =>
+         log.warning("Unable to send message {}", failedMsg)
+     }
+   }
+
+Using the ASK pattern, in this case it is useful to be able to map circuit open failures to the same type of failures
+returned by the service (a `Left[String]` in our case): ::
+
+   class CircuitBreakerAskExample(potentiallyFailingService: ActorRef) extends Actor with ActorLogging {
+     import SimpleService._
+     import akka.pattern._
+
+     implicit val askTimeout: Timeout = 2.seconds
+
+     val serviceCircuitBreaker =
+       context.actorOf(
+         CircuitBreakerActorBuilder( maxFailures = 3, callTimeout = askTimeout, resetTimeout = 30.seconds )
+           .copy(
+             failureDetector = {
+               _ match  {
+                 case Response(Left(_)) => true
+                 case _ => false
+               }
+             }
+           )
+           .copy(
+             openCircuitFailureConverter = { failure =>
+               Left(s"Circuit open when processing ${failure.failedMsg}")
+             }
+           )
+           .propsForTarget(potentiallyFailingService),
+         "serviceCircuitBreaker"
+       )
+
+     import context.dispatcher
+
+     override def receive: Receive = {
+       case AskFor(requestToForward) =>
+         (serviceCircuitBreaker ? Request(requestToForward)).mapTo[Either[String, String]].onComplete {
+           case Success(Right(successResponse)) =>
+             //handle response
+             log.info("Got successful response {}", successResponse)
+
+           case Success(Left(failureResponse)) =>
+             //handle response
+             log.info("Got successful response {}", failureResponse)
+
+           case Failure(exception) =>
+             //handle response
+             log.info("Got successful response {}", exception)
+
+         }
+
+     }
+   }
+
+
+If it is not possible to define define a specific error response, you can map the Open Circuit notification into a failure.
+That also means that your `CircuitBreakerActor` will be essentially useful to protect you from time out for extra workload or
```

Comment:
`will be essentially useful to protect` please remove essentially - just useful is enough